### PR TITLE
Add the mono DNU package and give it execute rights.

### PR DESCRIPTION
**Pull request by JeremyKuhne**
@ericstj, @ValMenn, @weshaggard 

I assume we should mirror the package in our own myget feed as we do for the other- @ericstj can you help me with that?

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by richlander** [comment]
Curious. What's the rationale for doing?

**Comment by JeremyKuhne** [comment]
@richlander Necessary to build on Unix for now.

**Comment by richlander** [comment]
I see. Yes, I noticed that. Looking forward to that going away.

File: src/NuGet.Config [hunk 1]
```diff
@@ -9,9 +9,10 @@
     <add key="myget.org dotnet-corefx" value="https://www.myget.org/F/dotnet-corefx/" />
     <add key="myget.org dotnet-corefxtestdata" value="https://www.myget.org/F/dotnet-corefxtestdata/" />
     <add key="myget.org dotnet-buildtools" value="https://www.myget.org/F/dotnet-buildtools/" />
+    <add key="myget-org ASP.NET vnext" value="https://www.myget.org/F/aspnetvnext" />
```
Comment (ericstj) [inline src/NuGet.Config]: Please remove this before merging.

**Comment by ericstj** [comment]
I gave you access to the buildtools feed.

File: src/.nuget/packages.config [hunk 2]
```diff
@@ -3,4 +3,5 @@
   <package id="Microsoft.DotNet.BuildTools" version="1.0.25-prerelease-00040" />
   <package id="dnx-coreclr-win-x86" version="1.0.0-beta5-11580" />
   <package id="Microsoft.DotNet.BuildTools.ApiTools" version="1.0.0-prerelease" />
+  <package id="dnx-mono" version="1.0.0-beta5-11596" />
```
Comment (ericstj) [inline src/.nuget/packages.config]: Bummer that we have to pull this down even for windows builds.  Did you consider splitting it out?

Comment (JeremyKuhne) [inline src/.nuget/packages.config]: Yeah, I'll probably do that. If we continue to have diverged dependencies perhaps we should create platform package configs for each platform. I'm going to assume that we will ultimately converge everything for now...

Comment (ericstj) [inline src/.nuget/packages.config]: I'm not sure when nuget team plans to deliver a cross-plat nuget.exe that supports v3 restore.  It's not in the super near future AFAIK so we'll continue to use DNX for at least the next few weeks.  I'd split it.

Comment (JeremyKuhne) [inline src/.nuget/packages.config]: I'll split it in a separate change after I get the other build issues worked out. To be agnostic I'll split out the Windows only package as well.

Comment (ericstj) [inline src/.nuget/packages.config]: Ok, once you get a clean build without nuget.config change go ahead and merge this.

**Pull request merged**